noir-projects/noir-protocol-circuits/crates/reset-kernel-lib/src/reset/transient_data.nr

```diff
@@ -70,8 +65,14 @@ pub fn verify_squashed_transient_note_hashes_and_nullifiers<NUM_NOTE_HASHES, NUM
     let mut note_logs_removed = 0;
     for i in 0..NUM_LOGS {
         let log = note_logs[i];
-        let index_hint = transient_note_hash_indexes_for_logs[i];
-        if index_hint == NUM_NOTE_HASHES {
+        let index_hint = transient_or_propagated_note_hash_indexes_for_logs[i];
+        // - For non-empty logs, their note_hash_counter can't be 0. It must match the counter of a note hash, which is verified in private_call_data_validator.
+        //   Therefor, is_propagated will be false if index_hint incorrectly points to an empty expected_note_hash and fails the check inside the else statement.
+        // - For empty logs, we expect is_propagated to be true, making them part of the note_logs_kept.
+        //   index_hint might point them to the same empty note. This doesn't matter because we increment the value of 
+        //   note_logs_kept and ensure that the correct number of empty logs are present in the expected_note_logs.
+        let is_propagated = expected_note_hashes[index_hint].counter() == log.note_hash_counter;
```

Comment:
The critical change to make sure all the note logs linked to a note hash must be removed if the note hash is removed.